src/tools/rust-analyzer/crates/hir-ty/src/utils.rs

```diff
@@ -334,6 +334,7 @@ pub(crate) fn detect_variant_from_bytes<'a>(
     e: EnumId,
 ) -> Option<(EnumVariantId, &'a Layout)> {
     let (var_id, var_layout) = match &layout.variants {
+        hir_def::layout::Variants::Empty => unreachable!(),
```

Comment:
A comment here would be nice. Took me a while to notice why it is unreachable.

Reply:
I don't know this code, and I don't know why it is unreachable. But this code would already have panicked before my PR if it got called with a variantless enum, so my patch just preserves those panics.

Reply:
I think getting some bytes seems to be a witness that the type in inhabited. I am not 100% sure what to think about empty slice.